File: BaseLibrary/command/common/WaitCommand.cs
using System.Collections.Generic;

namespace BaseLibrary.command.common {

    /// <summary>
    /// Wait one turn or to until HitPoint > 0 or to the end of turn
    /// </summary>
    public class WaitCommand : ACommonCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        public WaitCommand() : base() {
            pending = false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/common/WaitCommand.java

package BaseLibrary.communication.command.common;

import BaseLibrary.communication.protocol.AProtocol;

/**
 * Wait one turn or to until HitPoint > 0 or to the end of turn
 */
public class WaitCommand extends ACommonCommand {

	private static final String NAME = "WAIT";

	static {
		AProtocol.registerForDeserialize(NAME, WaitCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}

	public WaitCommand() {
		pending = false;
	}
}
